pyoffers/models/affiliate.py

```diff
@@ -0,0 +1,85 @@
+# coding: utf-8
+from .core import Model, ModelManager, RelatedManager
+
+AFFILIATE_USER_PARAMS = (
+    'access',
+    'cell_phone',
+    'email',
+    'first_name',
+    'last_name',
+    'password',
+    'password_confirmation',
+    'phone',
+    'status',
+    'title',
+    'wants_alerts'
+)
+
+
+class AffiliateUser(Model):
+    generic_methods = ('update', 'delete')
+
+
+class AffiliateUserManager(ModelManager):
+    model = AffiliateUser
+    name = 'affiliate_users'
+    generic_methods = (
+        'create',
+        'update',
+        'delete',
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+    )
+
+
+class RelatedUserManager(RelatedManager):
+    related_object_name = 'affiliate_id'
+
+    def get(self):
+        try:
+            return self.base_manager.find_all(**{self.related_object_name: self.id})[0]
+        except IndexError:
+            return None
+
+
+class Affiliate(Model):
+    """
+    An Affiliate.
+    """
+    generic_methods = ('update', 'delete')
+
+    @property
+    def affiliate_user(self):
+        return RelatedUserManager(api=self._manager.api, base_manager_class=AffiliateUserManager, id=self.id).get()
+
+    def block(self, reason=None):
+        return self._manager.block(self.id, reason=reason)
+
+
+class AffiliateManager(ModelManager):
+    model = Affiliate
+    name = 'affiliates'
+    generic_methods = (
+        'update',
+        'delete',
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+    )
+
+    def create(self, create_user=True, **kwargs):
```

Comment:
Let's simplify the interface and remove creating of AffiliateUser. It still will be possible via ``api.affiliate_users.create``